python/pyspark/sql/types.py

```diff
@@ -681,6 +681,129 @@ def __eq__(self, other):
                           for v in [ArrayType, MapType, StructType])
 
 
+_FIXED_DECIMAL = re.compile("decimal\\(\\s*(\\d+)\\s*,\\s*(\\d+)\\s*\\)")
+
+
+_BRACKETS = {'(': ')', '[': ']', '{': '}'}
+
+
+def _parse_basic_datatype_string(s):
+    if s in _all_atomic_types.keys():
+        return _all_atomic_types[s]()
+    elif s == "int":
+        return IntegerType()
+    elif _FIXED_DECIMAL.match(s):
+        m = _FIXED_DECIMAL.match(s)
```

Comment:
I think so, the code is copied from https://github.com/apache/spark/blob/master/python/pyspark/sql/types.py#L860-L864, we can improve them later